### DISCUSSION TITLE
[me.lucko.luckperms.lib.mariadb.Configuration] `useSsl` option is deprecated, replaced by option `sslMode`

### DISCUSSION THREAD

**Discussion by cwchristerw** [LuckPerms/LuckPerms#3654] [General]
I didn't find anything about sslMode in https://luckperms.net/wiki/Storage-system-errors

**Comment by cwchristerw** [comment]
I have used useSSL: true and verifyServerCertificate: false for many years now but now it gives me "[me.lucko.luckperms.lib.mariadb.plugin.tls.main.DefaultTlsSocketPlugin] IPv4 host "10.254.1.xxx" doesn't correspond to certificate CN" error

**Reply by cwchristerw** [reply]
In my case I should use sslMode: REQUIRED (https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-connp-props-security.html)

**Comment by RalphORama** [comment]
I had this issue too. I think `data.pool-settings.properties` in config.yml can take arbitrary connection string values, so all you have to do is add a line that reads `sslMode: <mode>`

From [HikariConnectionFactory.java](https://github.com/LuckPerms/LuckPerms/blob/master/common/src/main/java/me/lucko/luckperms/common/storage/implementation/sql/connection/hikari/HikariConnectionFactory.java#L99)

```java
    /**
     * Sets the given connection properties onto the config.
     *
     * @param config the hikari config
     * @param properties the properties
     */
    protected void setProperties(HikariConfig config, Map<String, Object> properties) {
        for (Map.Entry<String, Object> property : properties.entrySet()) {
            config.addDataSourceProperty(property.getKey(), property.getValue());
        }
    }
```

my current `config.yml` (note that I am using MariaDB):

```yaml
    properties:
      useUnicode: true
      characterEncoding: utf8
      sslMode: verify-full
      #useSSL: true
      #verifyServerCertificate: true
```
